packages/mobile/src/components/publish-content-drawer/PublishContentDrawer.tsx

```diff
@@ -90,15 +123,22 @@ export const PublishPlaylistDrawer = () => {
     )
 
     return toast({ content: publishingPlaylistToastContent })
-  }, [toast, styles])
+  }, [styles.toastContainer, styles.spinner, toast])
 
   const handlePublish = useCallback(() => {
-    if (playlistId) {
-      const { key: dismissToastKey } = displayPublishToast()
-      dispatch(publishPlaylist(playlistId, dismissToastKey))
-      onClose()
+    // Publish playlist
+    const { key: dismissToastKey } = displayPublishToast()
+    setDismissToastKey(dismissToastKey)
+
+    if (contentId && contentType === 'playlist') {
+      dispatch(publishPlaylist(contentId, dismissToastKey))
```

Comment:
Pre-existing logic in native. It already handled dismissing the toast inside the saga